apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data.Tests/CollectionReferenceTest.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright 2017, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Xunit;
+
+namespace Google.Cloud.Firestore.Data.Tests
+{
+    public class CollectionReferenceTest
+    {
+        [Fact]
+        public void Equality()
+        {
+            var client = new FakeFirestoreClient();
+            var db1 = FirestoreDb.Create("proj", "db", client);
+            var db2 = FirestoreDb.Create("proj", "db", client);
+            var db3 = FirestoreDb.Create("proj", "otherdb", client);
+
+            // All equal
+            var equal = new[]
+            {
+                db1.Collection("col1/doc1/col2"),
+                db1.Collection("col1/doc1/col2"),
+                db2.Collection("col1/doc1/col2"),
+                db1.Collection("col1").Document("doc1").Collection("col2")
+            };
+            var unequal = new[]
+            {
+                db1.Collection("col1x/doc1/col2"),
+                db1.Collection("col1/doc1x/col2"),
+                db1.Collection("col1/doc1/col2x"),
+                db3.Collection("col1/doc1/col2"),
+                db1.Collection("col1"),
+                db1.Collection("col2"),
+            };
+            EqualityTester.AssertEqual(equal, unequal);
```

Comment:
I was a bit confused by this and had to look into the implementation details to understand. I originally assumed everything in the `equal` collection had to compare equal to each other and everything int the `unequal` collection had to compare unequal to each other. Perhaps it would be a bit clearer to have the control passed it as a first argument. I think it would be more obvious that everything is being compared to that.

Reply:
I wondered about that - will do.